src/test/java/it/gov/pagopa/payhub/activities/service/paymentsreporting/PaymentsReportingMapperServiceTest.java

```diff
@@ -0,0 +1,90 @@
+		ctFlussoRiversamento.setCodiceBicBancaDiRiversamento("BIC123");
+		ctFlussoRiversamento.setNumeroTotalePagamenti(BigDecimal.valueOf(100L));
+		ctFlussoRiversamento.setImportoTotalePagamenti(BigDecimal.valueOf(1000.50d));
+
+		ctFlussoRiversamento.setDataRegolamento(toXMLGregorianCalendar(gregorianCalendar));
+		ctFlussoRiversamento.setDataOraFlusso(toXMLGregorianCalendar(gregorianCalendar));
+
+		CtIstitutoMittente istitutoMittente = new CtIstitutoMittente();
+		istitutoMittente.setDenominazioneMittente("PSP Mittente");
+		CtIdentificativoUnivoco identificativoMittente = new CtIdentificativoUnivoco();
+		identificativoMittente.setCodiceIdentificativoUnivoco("12345");
+		identificativoMittente.setTipoIdentificativoUnivoco(B);
+		istitutoMittente.setIdentificativoUnivocoMittente(identificativoMittente);
+		ctFlussoRiversamento.setIstitutoMittente(istitutoMittente);
+
+		CtIstitutoRicevente istitutoRicevente = new CtIstitutoRicevente();
+		istitutoRicevente.setDenominazioneRicevente("Org Ricevente");
+		CtIdentificativoUnivocoPersonaG identificativoRicevente = new CtIdentificativoUnivocoPersonaG();
+		identificativoRicevente.setCodiceIdentificativoUnivoco("54321");
+		identificativoRicevente.setTipoIdentificativoUnivoco(G);
+		istitutoRicevente.setIdentificativoUnivocoRicevente(identificativoRicevente);
+		ctFlussoRiversamento.setIstitutoRicevente(istitutoRicevente);
+
+		IngestionFlowFileDTO ingestionFlowFileDTO = new IngestionFlowFileDTO();
+		ingestionFlowFileDTO.setOrg(new OrganizationDTO());
+
+		CtDatiSingoliPagamenti singlePayment = new CtDatiSingoliPagamenti();
+		singlePayment.setIdentificativoUnivocoVersamento("vers123");
+		singlePayment.setIdentificativoUnivocoRiscossione("ris123");
+		singlePayment.setIndiceDatiSingoloPagamento(1);
+		singlePayment.setSingoloImportoPagato(BigDecimal.valueOf(200.00D));
+		singlePayment.setCodiceEsitoSingoloPagamento("OK");
+		singlePayment.setDataEsitoSingoloPagamento(toXMLGregorianCalendar(gregorianCalendar));
+
+		ctFlussoRiversamento.getDatiSingoliPagamenti().addAll(List.of(singlePayment));
+
+		// When
+		List<PaymentsReportingDTO> result = mapper.mapToDtoList(ctFlussoRiversamento, ingestionFlowFileDTO);
+
+		// Then
+		assertEquals(1, result.size());
+		assertEquals("flow123", result.get(0).getFlowIdentifierCode());
+		assertEquals("PSP Mittente", result.get(0).getSenderPspName());
+		assertEquals("Org Ricevente", result.get(0).getReceiverOrganizationName());
+		assertEquals(100L, result.get(0).getSumPayments().longValue());
+		assertEquals(200.00d, result.get(0).getAmountPaid().doubleValue());
+		assertEquals("vers123", result.get(0).getCreditorReferenceId());
+		assertEquals("ris123", result.get(0).getRegulationId());
+		assertEquals(1, result.get(0).getTransferIndex());
+		assertEquals(200.00D, result.get(0).getAmountPaid().doubleValue());
+		assertEquals("OK", result.get(0).getPaymentOutcomeCode());
+		assertNotNull(result.get(0).getPayDate());
+	}
```

Comment:
let's copy TestUtils.checkNotNullFields from other applications (eg. p4pa-auth) in order to check that all expected not null fields are filled